sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -94,6 +94,12 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     case Distinct(p: Project) =>
       projectToSQL(p, isDistinct = true)
 
+    case p@ Project(_, g: Generate) =>
+      generateToSQL(p)
+
+    case g: Generate =>
```

Comment:
@cloud-fan Sorry Wenchen for getting back to you a bit late. Actually the two cases do different things. First one builds the top level SELECT clause and the second one builds the LATERAL VIEW clause. Do you have any suggestion for a better name for these two functions ? :-)